src/Microsoft.Data.SqlClient/netcore/src/Microsoft/Data/SqlClient/SNI/SNIPhysicalHandle.cs

```diff
@@ -42,53 +42,33 @@ public override SNIPacket RentPacket(int headerSize, int dataSize)
                 Debug.Assert(packet.IsInvalid, "dequeue returned valid packet");
                 GC.ReRegisterForFinalize(packet);
             }
-#if TRACE_HISTORY
-            if (packet._history != null)
-            {
-                packet._history.Add(new SNIPacket.History { Action = SNIPacket.History.Direction.Rent, Stack = GetStackParts(), RefCount = packet._refCount });
-            }
-#endif
+            packet.AddHistory(true);
             Interlocked.Add(ref packet._refCount, 1);
-            Debug.Assert(packet.IsActive, "SNIPacket _refcount must be 1 or a lifetime issue has occurred, trace with the #TRACE_HISTORY define");
+            Debug.Assert(packet.IsActive, "SNIPacket _refcount must be 1 or a lifetime issue has occured, trace with the #TRACE_HISTORY define");
+            packet.ClearPath();
 #endif
             packet.Allocate(headerSize, dataSize);
             return packet;
         }
 
         public override void ReturnPacket(SNIPacket packet)
         {
-#if DEBUG
             Debug.Assert(packet != null, "releasing null SNIPacket");
-            Debug.Assert(packet.IsActive, "SNIPacket _refcount must be 1 or a lifetime issue has occurred, trace with the #TRACE_HISTORY define");
+#if DEBUG
+            Debug.Assert(packet.IsActive, "SNIPacket _refcount must be 1 or a lifetime issue has occured, trace with the #TRACE_HISTORY define");
             Debug.Assert(ReferenceEquals(packet._owner, this), "releasing SNIPacket that belongs to another physical handle");
-            Debug.Assert(!packet.IsInvalid, "releasing already released SNIPacket");
 #endif
+            Debug.Assert(!packet.IsInvalid, "releasing already released SNIPacket");
 
             packet.Release();
 #if DEBUG
             Interlocked.Add(ref packet._refCount, -1);
             packet._traceTag = null;
-#if TRACE_HISTORY
-            if (packet._history != null)
-            {
-                packet._history.Add(new SNIPacket.History { Action = SNIPacket.History.Direction.Return, Stack = GetStackParts(), RefCount = packet._refCount });
-            }
-#endif
+            packet.AddHistory(false);
+            //packet.PushPath(new StackTrace().ToString());
```

Comment:
Remove?

Reply:
Yes, I'd left it as an example but I've moved it to the comment in the packet.debug file instead.